### PR TITLE
Replace load-generator with Go implementation

### LINKED ISSUE #688: Rewrite load-generator in Go
The `/tools/load-generator` is the only component of this repo that is in Python. It would be nice to replace this with a Go binary in order to eliminate the Python ecosystem dependency for one tool. There are also no CI tests for this Python code, so we don't know if it's broken when Python or other requirements change.

### LINKED ISSUE #705: PromBench Idea: exercise more parts of PromQL
PromBench runs synthetic load defined in [6_loadgen.yaml](https://github.com/prometheus/test-infra/blob/34a189da3778d9035d430e3213f5c9dd18d520a7/prombench/manifests/prombench/benchmark/6_loadgen.yaml).
These cover selectors, `rate`, `histogram_quantile` and `sum`, but leave out most of PromQL.

For instance we could add:
* binary arithmetic operators (joins)
* logical operators (`and`, `or`, `unless`)
* `topk`

### PR DESCRIPTION
**Pull request by HarshitNagpal29**
* Replace python load-generator with a Go implementation.
* Add additional PromQL tests.

Fixes #688
Fixes #705

### CODE REVIEW

**Comment by HarshitNagpal29** [comment]
@SuperQ Sir, can you please check it once?

File: tools/load-generator/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+WORKDIR /usr/src/app
 
-COPY ./*.py /usr/src/app/
+COPY go.mod go.sum ./
+RUN go mod download
 
-WORKDIR /usr/src/app
+COPY . .
+
+RUN go build -o load-generator .
```
Comment (SuperQ) [inline tools/load-generator/Dockerfile]: We don't do builds in Docker here. What you want to do is add the new tool to the top level `.promu.yml`. Take a look at `tools/fake-webserver/Dockerfile` for an example.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ updates:
         patterns:
           - "go.opentelemetry.io/*"
     open-pull-requests-limit: 20
-  - package-ecosystem: "pip"
-    directory: "/tools/load-generator"
-    schedule:
-      interval: "monthly"
+  #- package-ecosystem: "pip"
```
Comment (SuperQ) [inline .github/dependabot.yml]: Please remove rather than comment out.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml [hunk 3]
```diff
@@ -48,6 +48,25 @@ data:
         - expr: histogram_quantile(0.99, sum by(path, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
         - expr: histogram_quantile(0.99, sum by(path, method, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
         - expr: histogram_quantile(0.99, sum by(instance, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
+      - name: binary_arithmetic_operators
```
Comment (SuperQ) [inline prombench/manifests/prombench/benchmark/6_loadgen.yaml:51]: This change seems unrelated.

File: tools/load-generator/main.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Println("unexpected arguments")
+		fmt.Println("usage: <load_generator> <namespace> <pr_number>")
+		os.Exit(2)
+	}
+
+	namespace = os.Args[1]
```
Comment (SuperQ) [inline tools/load-generator/main.go]: This appears unused.

**Review by SuperQ — CHANGES_REQUESTED** [review]

**Comment by SuperQ** [comment]
This needs a DCO sign-off. You can use `git commit -s --amend` to add it.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml (hunk 3 above)
Comment (SuperQ) [reply prombench/manifests/prombench/benchmark/6_loadgen.yaml:51]: Ahh, this is for #705

**Comment by HarshitNagpal29** [comment]
@SuperQ Sir first of all really sorry for not clarifying properly about PR, yes it resolved 2 issues this one and #705. I have a seperate PR for #705 as well but no was reviewing it.

Also the changes you suggested I will make in few hour and commit changes, and will see why tests are failing, and try to resolve it.

**Comment by SuperQ** [comment]
We generally don't review if tests are failing.

**Comment by HarshitNagpal29** [comment]
Also sir can you look for issue #594 once, like I have proposed it's solution and not created PR, I have mentioned everything there once you can check I have ready solution in python I will create it in go aswell.
@SuperQ

**Comment by HarshitNagpal29** [comment]
@SuperQ Sir i have made the changes as suggested by you, PTAl.

**Comment by HarshitNagpal29** [comment]
i dont know why DCO is failing even after git commit -s --amend

**Comment by SuperQ** [comment]
Each individual commit must be signed off. You can use `git rebase -i origin/master` to squash your commits.

**Comment by HarshitNagpal29** [comment]
@SuperQ Hi Sir, hope you are doing well, I was having multiple issues even after rebasing so I have created a new PR https://github.com/prometheus/test-infra/pull/743 signing and it has passed DCO, please review and approve the changes.

**Comment by HarshitNagpal2901** [comment]
@SuperQ Sir can I close this PR? and you can merge another one, is it fine?

**Comment by SuperQ** [comment]
Superseded by #748

**Pull request closed without merging**